datastore-design/go-grpc-demo/storagebackend/timescaledb/getobservations.go

```diff
@@ -4,37 +4,39 @@ import (
 	"database/sql"
 	"datastore/datastore"
 	"fmt"
-
 	_ "github.com/lib/pq"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"math"
 )
 
 // retrieveObsForTS retrieves into obs observations for time series tsID in
 // open-ended range [fromTime, toTime>.
 // Returns nil upon success, otherwise error.
 func retrieveObsForTS(
-	db *sql.DB, tsID int64, fromTime, toTime int64, obs *[]*datastore.Observation) error {
+	db *sql.DB, tsID int64, fromTime *timestamppb.Timestamp, toTime *timestamppb.Timestamp, obs *[]*datastore.Observation) error {
 
-	rows, err := db.Query(fmt.Sprintf(`
-		SELECT EXTRACT(EPOCH FROM tstamp)::INT, value, field1, field2 FROM observations
-		WHERE ts_id = %d
-		AND tstamp >= to_timestamp(%d)
-		AND tstamp <  to_timestamp(%d)
+	rows, err := db.Query(`
+		SELECT EXTRACT(EPOCH FROM tstamp), value, field1, field2 FROM observations
+		WHERE ts_id = $1
+		AND tstamp >= to_timestamp($2)
+		AND tstamp <  to_timestamp($3)
 		ORDER BY tstamp ASC
-	`, tsID, fromTime, toTime))
+	`, tsID, float64(fromTime.Seconds)+float64(fromTime.Nanos)/1e9, float64(toTime.Seconds)+float64(toTime.Nanos)/1e9)
```

Comment:
Are we required to represent sub-second obs times? If not, x.Nanos don't need to be added.

Reply:
I don't know if that is explicitly stated, but we might as well to make it future proof.

Reply:
Agreed. Not a big deal, really. (side note: for the internal observation storage system at MET we decided to restrict to seconds resolution to follow the [KISS princible](https://en.wikipedia.org/wiki/KISS_principle) as much as possible, and then generalize if the need would arise in the future)